src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -111,11 +116,22 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
           this.props.onSelectedDataSources([{ id: optionId, label: selectedDataSource.title }]);
         }
       } catch (error) {
-        handleDataSourceFetchError(
-          this.onError.bind(this),
-          this.props.notifications!,
-          this.props.onSelectedDataSources
-        );
+        // return [] to customer when no default data source
+        if (!defaultDataSource) {
+          this.props.onSelectedDataSources?.([]);
+        }
+        this.setState({
+          defaultDataSource: this.getFilteredDefaultDataSource(defaultDataSource),
```

Comment:
why do we need to set default data source here?

Reply:
Because when we hit the error state, we never save the defaultDataSource to some local state, functions out side of componentDidMount cannot use it